specification/cdn/resource-manager/Microsoft.Cdn/stable/2019-04-15/cdn.json

```diff
@@ -3230,6 +3231,109 @@
         }
       }
     },
+    "DeliveryRuleUrlSigningAction": {
+      "description": "Defines the Url Signing action for the delivery rule.",
+      "x-ms-discriminator-value": "UrlSigning",
+      "allOf": [
+        {
+          "$ref": "#/definitions/DeliveryRuleAction"
+        }
+      ],
+      "required": [
+        "parameters"
+      ],
+      "properties": {
+        "parameters": {
+          "description": "Defines the parameters for the action.",
+          "$ref": "#/definitions/UrlSigningParameters"
+        }
+      }
+    },
+    "UrlSigningParameters": {
+      "description": "Defines the parameters for the Url Signing action.",
+      "required": [
+        "keys",
+        "hmacAlgorithm",
+        "queryStringParams",
+        "useIpInHash"
+      ],
+      "keys": {
+        "description": "Keys to be used for verifying the hash",
+        "type": "array",
+        "items": {
+          "$ref": "#/definitions/UrlSigningKey"
+        }
+      },
+      "hmacAlgorithm": {
+        "description": "Algorithm to use for URL signing",
+        "type": "string",
+        "enum": [
+          "SHA256",
+          "SHA1"
```

Comment:
Have support only for SHA256.

Reply:
Resolved in new PR:

---
In reply to: [302752636](https://github.com/Azure/azure-rest-api-specs/pull/6438#discussion_r302752636) [](ancestors = 302752636)